src/dc_context.c

```diff
@@ -307,6 +336,45 @@ char* dc_get_blobdir(const dc_context_t* context)
  * INI-handling, Information
  ******************************************************************************/
 
+static int is_settable_config_key(const char* key)
+{
+	for (int i = 0; i < str_array_len(config_keys); i++) {
+		if (strcmp(key, config_keys[i]) == 0) {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+
+static int is_gettable_config_key(const char* key)
+{
+	for (int i = 0; i < str_array_len(sys_config_keys); i++) {
+		if (strcmp(key, sys_config_keys[i]) == 0) {
+			return 1;
+		}
+	}
+	return is_settable_config_key(key);
+}
+
+
+static char* get_config_keys_str() {
+	dc_strbuilder_t  ret;
+	dc_strbuilder_init(&ret, 0);
+
+	for (int i = 0; i < str_array_len(config_keys); i++) {
+		dc_strbuilder_cat(&ret, config_keys[i]);
+		dc_strbuilder_cat(&ret, " ");
+	}
+
+	for (int i = 0; i < str_array_len(sys_config_keys) - 1; i++) {
+		dc_strbuilder_cat(&ret, sys_config_keys[i]);
+		dc_strbuilder_cat(&ret, " ");
+	}
+	dc_strbuilder_cat(&ret, sys_config_keys[str_array_len(sys_config_keys) - 1]);
```

Comment:
just nitpicking:

i think, this is to avoid the last space.
if, for any reasons, sys_config_keys is empty in the future, this will cause reading invalid memory. currently, this will happen never, but maybe it is just easier to add all items in the loop and add the space only by checking `ret.buf[0]!=0`in both loops.

@bpiddubnyi what do you think? makes sense?